tox/session.py

```diff
@@ -673,12 +672,12 @@ def report_env(e):
 
     def info_versions(self):
         versions = ['tox-%s' % tox.__version__]
-        try:
-            version = py.process.cmdexec("virtualenv --version")
-        except py.process.cmdexec.Error:
-            versions.append("virtualenv-1.9.1 (vendored)")
-        else:
-            versions.append("virtualenv-%s" % version.strip())
+        proc = subprocess.Popen(
+            (sys.executable, '-m', 'virtualenv', '--version'),
+            stdout=subprocess.PIPE,
+        )
+        out, _ = proc.communicate()
+        versions.append('virtualenv-{0}'.format(out.decode('UTF-8').strip()))
```

Comment:
I personally don't like the .format way of string formatting, but as long as no one is forcing me to use it myself I can live with it :)

Reply:
interesting! I prefer it over `%` formatting 😆

Reply:
@obestwalter Maybe https://pyformat.info/ will convince you otherwise? :wink:

Reply:
@The-Compiler I am afraid not. I don't see any value in using .format for **simple** string formatting. So for now I will still choose

    'some text %s bla bla' % somevar

over

    'some text {0} bla bla'.format(somevar)

...  until I know better.

I also wish that Guido had engaged his time machine and introduced f-strings right away rather than in 3.6 ...

Reply:
with ~clever~ evil hacks, I could probably get this to work in python2.7

```
# -*- encoding: future_fstrings -*-
thing = 'world'
print(f'hello {thing}')
```

Reply:
Can't wait to drop 2.6 then :D

Reply:
It is done! https://github.com/asottile/future-fstrings

Reply:
Hah! Oh dear! 😆

Reply:
O.M.G.

Reply:
@asottile any performance hit compared to format?

Reply:
They'll be slower that native f string support but this hack will have identical performance to .format (it translates the f string into format calls in fact). Native f strings are only faster because they are implemented directly in byte code (which a backport can't quite do)